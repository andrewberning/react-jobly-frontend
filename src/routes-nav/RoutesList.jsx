import { Routes, Route, Navigate } from "react-router-dom";
import Homepage from "../homepage/Homepage";
import CompanyList from "../companies/CompanyList";
import JobList from "../jobs/JobList";
import CompanyDetail from "../companies/CompanyDetail";
import LoginForm from "../auth/LoginForm";
import ProfileForm from "../profiles/ProfileForm";
import SignupForm from "../auth/SignupForm";
import PrivateRoute from "./PrivateRoute";

export default function RoutesList({ login, signup }) {
  return (
      <Routes>
        <Route path="/" element={<Homepage />} />
        <Route path="/login" element={<LoginForm login={login} />} />
        <Route path="/signup" element={<SignupForm signup={signup} />} />
        <Route element={<PrivateRoute />}>
          <Route path="/companies" element={<CompanyList />} />
          <Route path="/jobs" element={<JobList />} />
          <Route path="/companies/:handle" element={<CompanyDetail />} />
          <Route path="/profile" element={<ProfileForm />} />
        </Route>
        <Route path="/*" element={<Navigate to="/"/>} />
      </Routes>
  )
}