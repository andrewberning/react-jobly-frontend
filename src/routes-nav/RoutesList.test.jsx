import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { UserProvider } from "../testUtils";
import RoutesList from "./RoutesList";

it("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <RoutesList />
        </UserProvider>
      </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <RoutesList />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});