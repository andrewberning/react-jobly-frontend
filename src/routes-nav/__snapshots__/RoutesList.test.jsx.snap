// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        Jobly
      </h1>
      <p
        class="lead"
      >
        All the jobs in one, convenient place.
      </p>
      <h2>
        Welcome Back, testuser!
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
